cmd/snap/cmd_snap_op.go

```diff
@@ -252,10 +252,25 @@ func (mx *channelMixin) setChannelFromCommandline() error {
 		mx.Channel = ch.chName
 	}
 
-	if !strings.Contains(mx.Channel, "/") && mx.Channel != "" && mx.Channel != "edge" && mx.Channel != "beta" && mx.Channel != "candidate" && mx.Channel != "stable" {
-		// shortcut to jump to a different track, e.g.
-		// snap install foo --channel=3.4 # implies 3.4/stable
-		mx.Channel += "/stable"
+	if mx.Channel != "" {
+		if ch, err := channel.Parse(mx.Channel, ""); err != nil {
+			full, er := channel.Full(mx.Channel)
+			if er != nil {
+				// the parse error has more detailed info
+				return err
+			}
+
+			// TODO: get escapes in here so we can bold the Warning
+			head := i18n.G("Warning:")
+			msg := i18n.G("Specifying a channel %q is relying on undefined behaviour. Interpreting it as %q for now, but this will change.\n")
+			warn := fill(fmt.Sprintf(msg, mx.Channel, full), utf8.RuneCountInString(head)+1) // +1 for the space
+			fmt.Fprint(Stderr, head, " ", warn, "\n\n")
+			mx.Channel = full
```

Comment:
this means /stable means latest/stable,  not stable.  it's likely fine but making it clear

Reply:
yeap